sdk/src/main/java/io/opentdf/platform/sdk/Autoconfigure.java

```diff
@@ -0,0 +1,749 @@
+        }
+
+        public String authority() throws AutoConfigureException {
+            Pattern pattern = Pattern.compile("^(https?://[\\w./-]+)/attr/[^/\\s]*$");
+            Matcher matcher = pattern.matcher(url);
+            if (!matcher.find()) {
+                throw new AutoConfigureException("invalid type");
+            }
+            return matcher.group(1);
+        }
+
+        public String name() throws AutoConfigureException {
+            Pattern pattern = Pattern.compile("^https?://[\\w./-]+/attr/([^/\\s]*)$");
+            Matcher matcher = pattern.matcher(url);
+            if (!matcher.find()) {
+                throw new AutoConfigureException("invalid attribute");
+            }
+            try {
+                return URLDecoder.decode(matcher.group(1), StandardCharsets.UTF_8.name());
+            } catch (Exception e) {
+                throw new AutoConfigureException("invalid type");
+            }
+        }
+    }
+
+    // Utility class for an attribute value FQN.
+    public static class AttributeValueFQN {
+        private final String url;
+        private final String key;
+
+        public AttributeValueFQN(String url) throws AutoConfigureException {
+            Pattern pattern = Pattern.compile("^(https?://[\\w./-]+)/attr/(\\S*)/value/(\\S*)$");
+            Matcher matcher = pattern.matcher(url);
+            if (!matcher.find() || matcher.group(1) == null || matcher.group(2) == null || matcher.group(3) == null) {
+                throw new AutoConfigureException("invalid type: attribute regex fail for [" + url + "]");
+            }
+
+            try {
+                URLDecoder.decode(matcher.group(2), StandardCharsets.UTF_8.name());
+                URLDecoder.decode(matcher.group(3), StandardCharsets.UTF_8.name());
+            } catch (Exception e) {
+                throw new AutoConfigureException("invalid type: error in attribute or value");
+            }
+
+            this.url = url;
+            this.key = url.toLowerCase();
+        }
+
+        @Override
+        public String toString() {
+            return url;
+        }
+        
+        public boolean equals(Object obj) {
+            if (this == obj) {
+                return true;
+            }
+            if (obj == null || !(obj instanceof AttributeValueFQN)) {
+                return false;
+            }
+            AttributeValueFQN afqn = (AttributeValueFQN) obj;
+            if ((this.url.equals(afqn.url)) && (this.key.equals(afqn.key))){
+                return true;
+            }
+            return false;
+        }
+
+        public String getKey() {
+            return key;
+        }
+
+        public String authority() throws AutoConfigureException {
+            Pattern pattern = Pattern.compile("^(https?://[\\w./-]+)/attr/\\S*/value/\\S*$");
+            Matcher matcher = pattern.matcher(url);
+            if (!matcher.find()) {
+                throw new AutoConfigureException("invalid type");
```

Comment:
This regex is validated in the constructor so it can't throw here.

```suggestion
        public String authority() {
            Pattern pattern = Pattern.compile("^(https?://[\\w./-]+)/attr/\\S*/value/\\S*$");
            Matcher matcher = pattern.matcher(url);
            if (!matcher.find()) {
                throw new RuntimeException("invalid attribute");
```